source/routing/raptor_api.cpp

```diff
@@ -1167,74 +1166,131 @@ void make_response(navitia::PbCreator& pb_creator,
+            }
+            else if (destinations->size() == 0) {
+                pb_creator.fill_pb_error(pbnavitia::Error::no_destination,
+                                         pbnavitia::NO_DESTINATION_POINT,
+                                         "no destination point");
+            }
         }
+        return;
     }
+
     // classical case : We have departures and destinations
-    else {
 
-        DateTime bound = clockwise ? DateTimeUtils::inf : DateTimeUtils::min;
-        typedef boost::optional<navitia::time_duration> OptTimeDur;
-        const auto direct_path = get_direct_path(worker, origin, destination);
-        const OptTimeDur direct_path_dur = direct_path.path_items.empty() ?
-            OptTimeDur() :
-            OptTimeDur(direct_path.duration / origin.streetnetwork_params.speed_factor);
+    // Compute direct path
+    typedef boost::optional<navitia::time_duration> OptTimeDur;
+    const auto direct_path = get_direct_path(worker, origin, destination);
+    const OptTimeDur direct_path_dur = direct_path.path_items.empty() ?
+        OptTimeDur() :
+        OptTimeDur(direct_path.duration / origin.streetnetwork_params.speed_factor);
 
-        for(bt::ptime datetime : datetimes) {
-            int day = (datetime.date() - raptor.data.meta->production_date.begin()).days();
-            int time = datetime.time_of_day().total_seconds();
-            DateTime init_dt = DateTimeUtils::set(day, time);
-
-            if(max_duration != std::numeric_limits<uint32_t>::max()) {
-                if (clockwise) {
-                    bound = init_dt + max_duration;
-                } else {
-                    bound = init_dt > max_duration ? init_dt - max_duration : 0;
-                }
+    // For each date time
+    DateTime bound = clockwise ? DateTimeUtils::inf : DateTimeUtils::min;
+    for(bt::ptime datetime : datetimes) {
+
+        // Compute start time and Bound
+        DateTime day = (datetime.date() - raptor.data.meta->production_date.begin()).days();
+        DateTime time = datetime.time_of_day().total_seconds();
+        DateTime request_date_secs = DateTimeUtils::set(day, time);
+        if(max_duration != DateTimeUtils::inf) {
+            if (clockwise) {
+                bound = request_date_secs + max_duration;
+            } else {
+                bound = request_date_secs > max_duration ? request_date_secs - max_duration : 0;
             }
-            std::vector<Path> tmp = raptor.compute_all(
-                *departures, *destinations, init_dt, rt_level, transfer_penalty, bound, max_transfers,
+        }
+
+        // Call raptor
+        // note : Loop is for min_nb_journeys options
+        //
+        // min_nb_journeys options :
+        // Compute several loop until the number of journeys >= min_nb_journeys
+        // For each step, we fing best pathes,
+        // start to the lastest departure + 1(clockwise) or the lastest arrival - 1.
```

Comment:
the ~~lastest~~ latest departure + 1(clockwise) or the ~~lastest~~ earliest arrival - 1.